app/src/main/java/org/mozilla/focus/download/EnqueueDownloadTask.java

```diff
@@ -0,0 +1,148 @@
+package org.mozilla.focus.download;
+
+import android.app.Activity;
+import android.app.DownloadManager;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Environment;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.content.LocalBroadcastManager;
+import android.webkit.CookieManager;
+import android.webkit.URLUtil;
+import android.widget.Toast;
+
+import org.mozilla.focus.R;
+import org.mozilla.focus.utils.Constants;
+import org.mozilla.focus.web.Download;
+
+import java.lang.ref.WeakReference;
+import java.util.List;
+
+/**
+ * Use Android's Download Manager to queue this download.
+ */
+public class EnqueueDownloadTask extends AsyncTask<Void, Void, EnqueueDownloadTask.ErrorCode> {
+
+    private WeakReference<Activity> activityRef;
+    private Download download;
+    private String refererUrl;
+
+    public EnqueueDownloadTask(@NonNull Activity activity, @NonNull Download download, @Nullable String refererUrl) {
```

Comment:
Do we really need an activity? or just a context to provide the localized string from resources?

Why are we not passing these constructor parameters to execute(context, download, string)?

Reply:
The context are used in both `doInBackground` and `onPostExecute`. Besides, I would like it's an activity object to have some life-cycle checking (if user left the activity) at the beginning of the `onPostExecute`.